torch/autograd/anomaly_mode.py

```diff
@@ -67,6 +67,7 @@ class detect_anomaly(object):
 
     def __init__(self):
         self.prev = torch.is_anomaly_enabled()
+        warnings.warn("Warning: Anomaly Detection has been enabled")
```

Comment:
nit: No need to have "Warning:" as the printing will already say that.
Also maybe you want to mention why you warn here: "Anomaly Detection has been enabled. This will significantly slow down your code and should only be enabled for debugging."